lib/kamal/commands/registry.rb

```diff
@@ -11,4 +14,18 @@ def login
   def logout
     docker :logout, registry.server
   end
+
+  def setup
+    combine \
+      docker(:start, "kamal-docker-registry"),
+      docker(:run, "--detach", "-p", "127.0.0.1:#{local_port}:5000", "--name", "kamal-docker-registry", "registry:2"),
```

Comment:
Publishing the port on your local machine will make it available in the same network segment (see https://github.com/moby/moby/issues/45610). That's why I didn't go any further with this at the time.

I think we can mitigate this though by generating a password for the registry when we boot it.

Reply:
Hmm looks like a pr was merged a few days ago to resolve that. Perhaps in the next release of buildkit it'll be fixed for us?

Reply:
Oh awesome! This will be great 🎉

Reply:
Looks like this is now released: https://docs.docker.com/engine/release-notes/28/